File: cjhirashi-career-admin/src/types/adminSections.ts
export type AdminSectionType = 'table' | 'functional' | 'metrics' | 'bucket'

export const ADMIN_SECTION_TYPE_LABEL: Record<string, string> = {
  table: 'tabla',
  functional: 'funcional',
  metrics: 'métricas',
  bucket: 'bucket',
}

export interface AdminSectionView {
  key: string
  label: string
  description: string
  sidebar_title: string
  /** Instrucciones del sidebar derecho (Markdown/GFM). '' = override vacío explícito. */
  sidebar_body: string
  is_default: boolean
}

export interface AdminSection {
  id: string
  system_name: string
  label: string
  path: string
  section_type: AdminSectionType | string
  group: string
  resource_key: string | null
  related_tools: string[]
  default_agent_profile_id: string | null
  /** Agente L2 del chat contextual del sidebar derecho, o null = sin chat. */
  agent_profile_id: string | null
  agent_label: string | null
  agent_is_default: boolean
  sidebar_has_chat: boolean
  sidebar_has_instructions: boolean
  view_count: number
  views: AdminSectionView[]
}

export interface AdminSectionUpdate {
  agent_profile_id?: string | null
  views?: Record<string, Partial<Pick<AdminSectionView, 'description' | 'sidebar_title' | 'sidebar_body'>>>
}

export function matchAdminSection(
  pathname: string,
  sections: AdminSection[]
): { section: AdminSection; view: AdminSectionView } | null {
  const path = pathname.replace(/\/+$/, '') || '/'
  const exact = sections.find((section) => section.path === path)
  if (exact) {
    return { section: exact, view: exact.views[0] }
  }
  const prefixed = sections
    .filter((section) => path.startsWith(`${section.path}/`))
    .sort((a, b) => b.path.length - a.path.length)
  const section = prefixed[0]
  if (!section) return null
  const record = section.views.find((view) => view.key === 'view')
    ?? section.views.find((view) => view.key === 'record')
  return { section, view: record ?? section.views[section.views.length - 1] }
}

File: cjhirashi-career-api/src/schemas/admin_sections.py
"""Schemas del catálogo de secciones del Admin."""
from typing import Dict, List, Optional

from pydantic import BaseModel, Field


class AdminSectionView(BaseModel):
    key: str
    label: str
    description: str
    sidebar_title: str
    sidebar_body: str
    is_default: bool = True


class AdminSectionViewUpdate(BaseModel):
    description: Optional[str] = None
    sidebar_title: Optional[str] = None
    sidebar_body: Optional[str] = None


class AdminSectionItem(BaseModel):
    id: str  # PK sec-N (ADR-021)
    system_name: str  # slug legible: dashboard, career-projects…
    label: str
    path: str
    section_type: str
    group: str = ""
    resource_key: Optional[str] = None
    related_tools: List[str] = []
    default_agent_profile_id: Optional[str] = None
    # feature 001: agente L2 del chat contextual del sidebar derecho (o None).
    agent_profile_id: Optional[str] = None
    agent_label: Optional[str] = None
    agent_is_default: bool = True
    sidebar_has_chat: bool = False
    sidebar_has_instructions: bool = False
    view_count: int
    views: List[AdminSectionView]


class AdminSectionUpdateRequest(BaseModel):
    agent_profile_id: Optional[str] = Field(
        default=None,
        description=(
            "Agente L2 del chat contextual. Omitir para no cambiar. "
            "String vacío restaura el default de código."
        ),
    )
    views: Optional[Dict[str, AdminSectionViewUpdate]] = None
